### DISCUSSION TITLE
Asrock x299 Steel Legend Bios from support

### DISCUSSION THREAD

**Discussion by timido22** [xCuri0/ReBarUEFI#182] [Q&A]
Hello, I have a question regarding my computer's BIOS. I received a new BIOS from the support team 1.50L, but I'm unable to patch it. However, I have the latest BIOS version available on the official website 1.50, and I am able to patch it without any issues.

I'm confused about which BIOS version I should choose. Can you please advise me on the best option and also explain why I am unable to patch the BIOS received from the support team?

I have also attached an image for reference. Thank you.

![image](https://github.com/xCuri0/ReBarUEFI/assets/154282702/d0584e52-590f-44ba-a3f9-d8849d93af07)

**Comment by timido22** [comment]
I found bios 1.50(from website) without CAM(Clever Access Memory). Is it critical?

**Comment by timido22** [comment]
"I am unable to flash the patched 1.50 version as I am receiving an error message that says 'Secure Flash check fail'."

**Comment by MrAnonymous7122** [comment]
The screenshot in the original question is fine, the file is patched and therefore cannot be patched again. I don't think CAM is relevant to whether ReBAR will work or not. The error message "Secure Flash check fail" refers to your manufacturer's protection against BIOS files that are not their own.  One easy workaround is buying a CH341A Programmer(they are cheap and work well).

**Reply by timido22** [reply]
In the screenshot, you can see 2 different firmware for bios. I checked 3 bios before. 1,30G and 1.50L(with CAM) and 1.50(Without CAM). Only 1.50(without CAM) I could patch, other 2 with an error.

I understand why I see this error(Secure Flash). But I want to do this thing without a Programmer. Because I have a new one (bought it yesterday motherboard, and the price is over 300$ :) I don't want to void the warranty.

**Reply by AresLiCa** [reply]
> In the screenshot, you can see 2 different firmware for bios. I checked 3 bios before. 1,30G and 1.50L(with CAM) and 1.50(Without CAM). Only 1.50(without CAM) I could patch, other 2 with an error.
> 
> I understand why I see this error(Secure Flash). But I want to do this thing without a Programmer. Because I have a new one (bought it yesterday motherboard, and the price is over 300$ :) I don't want to void the warranty.

Did you end up resovling this? I got the same mother board recently and am in need of enabling CAM/rebar. Could you please share your experience further? Thank you@
